src/main/java/org/jabref/gui/fieldeditors/identifier/BaseIdentifierEditorViewModel.java

```diff
@@ -0,0 +1,142 @@
+            return Optional.empty();
+        }
+
+        identifier.set((Optional<T>) identifierParser.parse(field));
+        return identifier.get();
+    }
+
+    protected void handleIdentifierFetchingError(Exception exception, IdFetcher<T> fetcher) {
+        LOGGER.error("Error while fetching identifier", exception);
+        if (exception instanceof FetcherClientException) {
+            dialogService.showInformationDialogAndWait(Localization.lang("Look up %0", fetcher.getName()), Localization.lang("No data was found for the identifier"));
+        } else if (exception instanceof FetcherServerException) {
+            dialogService.showInformationDialogAndWait(Localization.lang("Look up %0", fetcher.getName()), Localization.lang("Server not available"));
+        } else if (exception.getCause() != null) {
+            dialogService.showWarningDialogAndWait(Localization.lang("Look up %0", fetcher.getName()), Localization.lang("Error occured %0", exception.getCause().getMessage()));
+        } else {
+            dialogService.showWarningDialogAndWait(Localization.lang("Look up %0", fetcher.getName()), Localization.lang("Error occured %0", exception.getCause().getMessage()));
+        }
+    }
+
+    public BooleanProperty canFetchBibliographyInformationByIdProperty() {
+        return canFetchBibliographyInformationById;
+    }
+
+    public boolean getCanFetchBibliographyInformationById() {
+        return canFetchBibliographyInformationById.get();
+    }
+
+    public BooleanProperty canLookupIdentifierProperty() {
+        return canLookupIdentifier;
+    }
+
+    public boolean getCanLookupIdentifier() {
+        return canLookupIdentifier.get();
+    }
+
+    public BooleanProperty isInvalidIdentifierProperty() {
+        return isInvalidIdentifier;
+    }
+
+    public boolean getIsInvalidIdentifier() {
+        return isInvalidIdentifier.get();
+    }
+
+    public boolean getIdentifierLookupInProgress() {
+        return identifierLookupInProgress.get();
+    }
+
+    public BooleanProperty identifierLookupInProgressProperty() {
+        return identifierLookupInProgress;
+    }
+
+    public void fetchBibliographyInformation(BibEntry bibEntry) {
+        LOGGER.warn("Unable to fetch bibliography information using the '{}' identifier", field.getDisplayName());
+    }
+
+    public void lookupIdentifier(BibEntry bibEntry) {
+        LOGGER.warn("Unable to lookup identifier for '{}'", field.getDisplayName());
+    }
+
+    public void openExternalLink() {
+        identifier.get().flatMap(Identifier::getExternalURI).ifPresent(url -> {
+                    try {
+                        JabRefDesktop.openBrowser(url);
+                    } catch (IOException ex) {
+                        dialogService.showErrorDialogAndWait(Localization.lang("Unable to open link."), ex);
+                    }
+                }
+        );
+    }
+
+    @Override
+    public void bindToEntry(BibEntry entry) {
```

Comment:
I'm really unsure about the `bindToEntry` method. It is called every time the tab in the entry editor the fieldeditor sits inside recieves focus (`EntryEditorTab::notifyAboutFocus`).
Maybe `EasyBind.listen` in the constructor (subscribe already runs the method on first call, listen only after the first change to the property).

Some hints at `AbstractEditorViewModel`

The entry editor is broken and needs to be rewritten. 😭

Reply:
`bindToEntry` is actually called the first time you select the tab. If you navigate away from the tab and back again, `bindToEntry` wouldn't be called. It would be called however if you select another entry, which makes sense.

About using `subscribe` instead of `listen`, it's necessary so that whenever the user switches to another entry, the UI state --which actions are visible on the side of that identifier editor-- is updated immediately to match the newly selected entry.